### PR TITLE
Use carbon-web-components directly in the ui-shell

### PR DESCRIPTION
**Pull request by abdonrd**
## Changes

After review https://github.com/Qiskit/web-components/pull/139 and https://github.com/Qiskit/web-components/pull/140

I think we can remove some elements in favor of using directly the original components.

We can't do it `header-menu` or the `header-menu-button` because they don't expose parts for the svgs.

## Implementation details

Just remove custom elements to use the original ones, editing the styles with the custom parts.

## How to read this PR

Commit by commit should be easier to read.

### CODE REVIEW

**Comment by abdonrd** [comment]
Friendly ping to @yoshiemuranaka as you lead the ui-shell!

File: components/ui-shell/index.scss [hunk 1]
```diff
@@ -62,30 +62,53 @@ $prefix: "bx";
...
+  &::before,
   &::part(divider) {
     display: none;
   }
 }
 
-:host(#{$q}-header-nav-item.#{$q}-user-accout-icon) {
-  a {
+#{$prefix}-header-nav-item.#{$q}-user-accout-icon {
+  &::part(link) {
```
Comment (yoshiemuranaka) [inline components/ui-shell/index.scss:81]: Noticing the account icon turn white on focus/hover. I think this should cover the states
Screen Shot 2022-09-09 at 7 24 50 AM



```suggestion
  &::part(link):focus,
  &::part(link):hover,
  &::part(link) {
```

File: components/ui-shell/index.scss [hunk 2]
```diff
@@ -108,15 +131,13 @@ $prefix: "bx";
...
-  /* stylelint-disable-next-line scss/at-extend-no-missing-placeholder */
-  @extend .#{$prefix}--header__menu-item;
-
+#{$prefix}-header-menu-item {
   height: 3rem;
   background-color: var(--cds-ui-01);
   color: var(--cds-text-secondary);
 
-  &::part(link) {
+  &::part(link),
```
Comment (yoshiemuranaka) [inline components/ui-shell/index.scss:139]: Also for the focus state in the dropdown
Screen Shot 2022-09-09 at 7 40 05 AM


```suggestion
  &::part(link),
  &::part(link):focus,
```

**Review by yoshiemuranaka — CHANGES_REQUESTED** [review]
Thanks for taking the time to do this so carefully @abdonrd!! 🎉 I love deleting code 😅  this looks much better!
Left comments for a few small issues with the focus/hover states

**Comment by yoshiemuranaka** [comment]
Also a question/comment about the community nav dropdown. I think when I started working on this, I referenced [learn.qiskit.org](https://learn.qiskit.org/account/learning), which doesn't include the `Ecosystem` dropdown but [qiskit.org](https://qiskit.org/) does. Maybe we can add `Ecosystem` with this PR? not sure whether or not it should be included in the final nav

**Comment by abdonrd** [comment]
Good catches! Thank you @yoshiemuranaka! 👏
I just applied your changes!

**Comment by abdonrd** [comment]
> Also a question/comment about the community nav dropdown. I think when I started working on this, I referenced [learn.qiskit.org](https://learn.qiskit.org/account/learning), which doesn't include the `Ecosystem` dropdown but [qiskit.org](https://qiskit.org/) does. Maybe we can add `Ecosystem` with this PR? not sure whether or not it should be included in the final nav

Hmmm I don't know...
And I can't find any information about it on Figma.

@JRussellHuffman could you confirm it? Thanks!

**Review by yoshiemuranaka — APPROVED** [review]

**Comment by abdonrd** [comment]
I merge it for now and I'm waiting for Russell's answer to do it in another PR.

**Pull request merged**